### PR TITLE
Rcs 38 passport auth

### PR DESCRIPTION
**Pull request by Astthor**
Authentication set up

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -1,18 +1,48 @@
...
 
 const port = process.env.APP_PORT || 5000;
 
 // Initialize the express engine
 const app: express.Application = express();
 
+// Always use a helmet (Security reasons: https://expressjs.com/en/advanced/best-practice-security.html)
+app.use(helmet());
+app.use(session({
+  secret: 'roll-call-legendary-app', // salt, make session id's harder to crack, value can be anything
```
Comment (DryDish) [inline src/app.ts]: Surely we can put the secret in a .env file? feels dirty to hard code it like this

File: src/authentication/user.authentication.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import { Request, Response, NextFunction } from "express";
+import passport from "passport";
+
+// Didn't work to have it in the global file.
+declare global {
+  namespace Express {
+    interface User {
+      username: string;
+      id?: number;
```
Comment (DryDish) [inline src/authentication/user.authentication.ts]: I may have an alternative fix for this, we talk on Thursday about it.

But like, in what corner of the dark web did you find this?

**Review by DryDish — CHANGES_REQUESTED** [review]
We look over it on thurdsday, looks good overall, just needs a couple tweaks

File: src/app.ts (hunk 1 above)
Comment (Astthor) [reply src/app.ts]: Haha yes, completely forgot about this one, it's named "secret" and I comment it as a thing to make sessions harder to crack. Then leave it exposed 🥇

File: src/authentication/user.authentication.ts (hunk 2 above)
Comment (Astthor) [reply src/authentication/user.authentication.ts]: Haha I think this is one of the recommended way of doing it, pretty sure it was one of the most rated answers on stack overflow for the error msg, it seems logical to me to extend the interface.
Either way would love to see an alternative solution!

File: src/routes/authentication-router.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+// Routing for: /auth/...
+// ------------------------------------------------
+
+// Authentication/Authorization route for microsoft (Our users don't navigate to this endpoint)
+router.get('/microsoft', passport.authenticate('microsoft', {
+  successRedirect: '/',
+  failureRedirect: '/auth/login-failed'
+}));
+
+router.get('/login-failed', (_req, res) => {
```
Comment (Teodor25) [inline src/routes/authentication-router.ts:16]: we need this? Could be removed i think

File: src/utils/populate-db.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  });
+  console.log('after building');
+  await studentRole.save();
+  console.log('after saving');
+  
+  const teacherRole = Role.build({
+    name: "teacher"
+  });
+  await teacherRole.save();
+
```
Comment (Teodor25) [inline src/utils/populate-db.ts]: why so many spaces? :)

File: src/utils/populate-db.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
+import { Role } from "../models/roles";
+import { User } from "../models/users";
+
+
```
Comment (Teodor25) [inline src/utils/populate-db.ts]: spaces?

**Review by Teodor25 — APPROVED** [review]

**Pull request merged**